partner_contact_sale_info_propagation/models/res_partner.py

```diff
@@ -1,50 +1,64 @@
 # Copyright 2019 Tecnativa - Ernesto Tejeda
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
 from odoo import api, models
 from lxml import etree
+from odoo.tools import config
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    def _check_propagation_allowed(self):
+        return bool(
+            not config["test_enable"]
+            or (
+                config["test_enable"]
+                and not self.env.context.get("prevent_propagation", False)
+            )
+        )
+
     @api.multi
     def write(self, vals):
         """Propagate Salesperson and Sales Channel change in the partner to the
         child contacts."""
-        for record in self:
-            if 'user_id' in vals:
-                childs = record.mapped('child_ids').filtered(
-                    lambda r: not r.user_id or r.user_id == record.user_id)
-                if childs:
-                    childs.write({'user_id': vals['user_id']})
-            if 'team_id' in vals:
-                childs = record.mapped('child_ids').filtered(
-                    lambda r: not r.team_id or r.team_id == record.team_id)
-                if childs:
-                    childs.write({'team_id': vals['team_id']})
-        return super(ResPartner, self).write(vals)
+        if self._check_propagation_allowed():
```

Comment:
Avoid this diff doing an early return:
```suggestion
        if not self._check_propagation_allowed():
            return super().write(vals)
```